File: lec02/lec02q2.cpp
//Printing prime numbers 

#include <iostream>
using namespace std;
int main(){

    int n;
    cin>> n;

    for (int cn=2; cn<n; cn++){
        bool isprime=true;
        for(int div=2; div*div<=cn; div++){
            if(cn%div==0){
                isprime= false;
                break;
            }
        }
            
            if(isprime == true){
                cout<< cn<<" ";
            }

        
    }
return 0;
}

File: lec02/lec02q2.java
//printing the prime numbers upto n

import java.util.Scanner;
class lec02q2{

    public static void main(String[] args){
         Scanner scn= new Scanner(System.in);

         int n= scn.nextInt();

         for(int i=2; i<n; i++){

            boolean isprime = true;
            
           for(int j=2; j*j<=i; j++){
            if(i%j==0){
                isprime=false;
                break;
            }
           }
        
            if(isprime==true){
                System.out.println(i);
                }
            
        }
    }
}
